### PR TITLE
chore: dependency updates and fixes

### LINKED ISSUE #106: chore: upgrade dependencies
While [draco2](https://github.com/cmudig/draco2) is being promising and actively developed, it is still a work in progress and cannot be fully integrated into a new application that would leverage ASP-based VisRec. Until then, it would be great to refresh this project to use more up-to-date dependencies.

### LINKED ISSUE #107: fix: word splitting in `js/concat_lp.sh`
Due to the fact that path variables are not quoted in the bash script responsible for generating `./js/src/constraints.ts`, word splitting may occur on some hosts. Quoting these variables would be a great improvement.

### LINKED ISSUE #108: fix: `tests/test_run.py`
Currently, the test `tests/test_run.py` is failing.

This is caused by the fact that `examples/data/cars.csv` includes `null` values, and when it is converted to a pandas `DataFrame` in `draco/helper.py` via `read_data_to_asp`, these null values get converted to `na` values by pandas by default.

This can be fixed easily by using:

```python
df = pd.read_csv(file, keep_default_na=False, na_values=None)
```

### LINKED ISSUE #109: Cannot create pull request
Hi,

I just addressed issues #106, #107 and #108 in my [fork](https://github.com/peter-gy/draco), however, I am not able to create a PR, since the corresponding button is disabled. Is this on purpose?

Screenshot 2022-07-25 at 11 30 42

### PR DESCRIPTION
**Pull request by peter-gy**
### Goals

- Update dependencies
  - use `vega-lite` `^5.3.0` -> produce output with schema `https://vega.github.io/schema/vega-lite/v5.json`
  - use `typescript` `^4.7.4`
  - use `rollup` `^2.77.0` -> update plugins to the non-archived node modules
- fix word splitting in `js/concat_lp.sh`
- fix `tests/test_run.py`
- enhance the docs of the `js` module
- update GH workflow to use `actions/setup-node@v3` and `node-version: 16` to install the fresh dependencies properly

### Notes

While [draco2](https://github.com/cmudig/draco2) is being promising and actively developed, it is still a work in progress and cannot be fully integrated into a new application that would leverage ASP-based VisRec. Until then, it would be great to refresh this project to use more up-to-date dependencies.

closes #106
closes #107
closes #108
closes #109 🙃

### CODE REVIEW

**Comment by domoritz** [comment]
Thank you for this pull request and all the other fixes you have sent. I'd love to learn more about what you are doing and what's missing from Draco 2.

**Comment by peter-gy** [comment]
> Thank you for this pull request and all the other fixes you have sent. I'd love to learn more about what you are doing and what's missing from Draco 2.

You are welcome; I am glad that my contributions to the repo were accepted.

I am actively working on [visrecl.ly](https://github.com/peter-gy/visrec.ly), a user-task-based visualization recommendation tool that happens to be the topic for my Bachelor's Thesis at Uni Wien. It is **heavily** inspired by your projects and papers, so thank you very much for your academic contributions!

The main reason I decided to depend on Draco in my project instead of Draco 2 is that - as outlined in cmudig/draco2#341 - the latter does not have a default spec-renderer at hand just yet. On the other hand, with Draco, I just need to plug the generated vega-lite specs into [react-vega](https://github.com/vega/react-vega), and the recommendations will appear automagically on the frontend (: 

Another compelling (but not decisive) argument for Draco was that its API is written in TypeScript, making it an excellent fit for my web-based stack. Also, as [clingo-wasm](https://github.com/domoritz/clingo-wasm) solves my ASP problems in the browser, the need for a Python-based rec server got eliminated; hence I will be able to explore offline PWA options as future work.

Long story short, a default renderer and a web API for Draco 2 would be very useful imho, due to the inherent nature of visualizations: most users want to play around with the results immediately, right in their browser.

**Comment by domoritz** [comment]
That's awesome. Yeah, I can see the appeal of a purely web based solution.

Note that I hope Draco 2 at least partially could also work on the web eventually. The constraints generated by the python program are pure asp and could run in clingo wasm. The python code is mostly there for testing and interfacing with higher-level programs. But if that interfacing is needed (as it seems in your case) Draco 1 makes more sense. Maybe we can whip up a pyscript based solution at some point.

**Comment by domoritz** [comment]
I'd to see a demo at some point. Maybe send me an email when you have something to show.

**Comment by peter-gy** [comment]
> I'd to see a demo at some point. Maybe send me an email when you have something to show.

Absolutely, I would appreciate your feedback a lot!
